// Message containing the updates for one simulation object for one
//  time quanta
//
// Rick Stovall, 2009-03-17
//
// This file released under the GPL 3.0 license.
// 
// Note: these are never sent bare.. they are always payloads 
// 	in a channel wrapper message.

package nrtb_msg;

import "physics_common.proto";
import "sim_management.proto";

// This message contains all the information required to report the
//  the status or status changes of any given object in a simulation
//  in any given time quanta.
// This message should never be sent alone. It's intended solely for 
//  use as member of the tq_data message defined below.
message sim_obj_tq_data {
    required uint32 obj_uid = 1;
    optional location_data location = 2;
    optional velocity_data velocity = 3;
    repeated string event = 4;
    repeated string inbound_comm = 5; 
    repeated string outbound_comm = 6;
}

// This is a container to hold all object status changes for a
//  given time quanta.
message tq_data {
    required uint32 quanta_id = 1; // at 20 ms, this allows 2.72 years!
    required uint32 sim_uid = 2;
    repeated sim_obj_tq_data obj_data = 3; 
	repeated obj_setup_data new_objs = 4;
}
